### ISSUE TITLE
inventory scale breaks with blur+ installed

### ISSUE THREAD

**Issue by MicrocontrollersDev** [inglettronald/DulkirMod-Fabric#13]
Blur+ mod: https://modrinth.com/mod/blur-fabric

screenshot: 
![image](https://github.com/user-attachments/assets/14c33a8d-7c9e-4839-8dcb-878ac37c6187)

**Comment by inglettronald** [comment]
Thank you for the report, I'll look into this soonish hopefully (by the latest when I bump to 1.21.whateverNextVersion)

**Comment by inglettronald** [comment]
Tbh I'm not gonna bother fixing this sorry. If anyone else wants to look into it feel free, but I do not have the time atm. This feature is by nature very invasive/conflicting with any other mod that would do GUI code modifications in the inventory, so It's likely to be broken with a bunch of things.

**Comment by MicrocontrollersDev** [comment]
for what its worth i was able to fix the background blur and have it fill up the entire screen with mixinsquared injections.

```java
    @TargetHandler(
            mixin = "eu.midnightdust.blur.mixin.MixinScreen",
            name = "blur$rotatedGradient"
    )
    @ModifyArg(
            method = "@MixinSquared:Handler",
            at = @At(
                    value = "INVOKE",
                    target = "Leu/midnightdust/blur/Blur;renderRotatedGradient(Lnet/minecraft/client/gui/DrawContext;II)V"
            ),
            index = 1
    )
    private int rotatedGradientWidth(int width) {
        return (int) ceil((double) width * InventoryScale.INSTANCE.getScale());
    }

    @TargetHandler(
            mixin = "eu.midnightdust.blur.mixin.MixinScreen",
            name = "blur$rotatedGradient"
    )
    @ModifyArg(
            method = "@MixinSquared:Handler",
            at = @At(
                    value = "INVOKE",
                    target = "Leu/midnightdust/blur/Blur;renderRotatedGradient(Lnet/minecraft/client/gui/DrawContext;II)V"
            ),
            index = 2
    )
    private int rotatedGradientHeight(int height) {
        return (int) ceil((double) height * InventoryScale.INSTANCE.getScale());
    }
    ```
    
i was not able to fix the issue of inventory positioning however so i never bothered PRing this half fix, id imagine its a similar change though. and even then im not sure how you feel about mixing into a mixin, its inherently brittle and a bandage at best.

**Comment by inglettronald** [comment]
That is quite interesting, thank you for the extra info!

**Comment by MicrocontrollersDev** [comment]
Hey, I finally managed to find a fix for this for Blur+ 5.2.0! Blur+ refactored quite heavily since my previous partial fix, and MixinSquared is no longer required. Do note this is using Mojmap and altered codebase, but the general idea is the same.

```java
@Mixin(value = Blur.class, remap = false)
public abstract class BlurMixin {
    @ModifyVariable(method = "renderRotatedGradient(Lnet/minecraft/client/gui/GuiGraphics;II)V", at = @At(value = "HEAD"), index = 1, argsOnly = true)
    private static int modifyWidth(int width) {
        return (int) (width * Scale.getScale());
    }

    @ModifyVariable(method = "renderRotatedGradient(Lnet/minecraft/client/gui/GuiGraphics;II)V", at = @At(value = "HEAD"), index = 2, argsOnly = true)
    private static int modifyHeight(int height) {
        return (int) (height * Scale.getScale());
    }

    @Inject(method = "renderRotatedGradient(Lnet/minecraft/client/gui/GuiGraphics;II)V", at = @At("HEAD"))
    private static void beforeGradient(GuiGraphics context, int width, int height, CallbackInfo ci) {
        context.pose().pushPose();
    }

    @Inject(method = "renderRotatedGradient(Lnet/minecraft/client/gui/GuiGraphics;II)V", at = @At("RETURN"))
    private static void afterGradient(GuiGraphics context, int width, int height, CallbackInfo ci) {
        context.pose().popPose();
    }
}
```
